src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -773,6 +781,12 @@ private static void SubstituteConstraintTypes(ImmutableArray<TypeWithAnnotations
                     return false;
                 }
 
+                if (considerParamsAndDefaultValues && (param1.IsParams != param2.IsParams ||
+                    param1.ExplicitDefaultConstantValue?.Value != param2.ExplicitDefaultConstantValue?.Value))
```

Comment:
You can just directly compare `ExplicitDefaultConstantValues`. This will be especially useful for long string values, as we won't realize the strings in memory, but will instead leave them as ropes.